tests/fuzzer.rs

```diff
@@ -0,0 +1,156 @@
+use flowgger;
+use quickcheck;
+
+use quickcheck::QuickCheck;
+
+use flowgger::flowgger::config::Config;
+use flowgger::flowgger::encoder::Encoder;
+use flowgger::flowgger::decoder::Decoder;
+use flowgger::flowgger::merger;
+use flowgger::flowgger::output;
+
+use std::sync::mpsc::{Receiver, sync_channel, SyncSender};
+
+use flowgger::flowgger::get_decoder_rfc3164;
+use flowgger::flowgger::get_encoder_rfc3164;
+use flowgger::flowgger::input::udp_input::handle_record_maybe_compressed;
+
+use self::merger::{LineMerger, Merger};
+use self::output::FileOutput;
+use self::output::Output;
+
+use std::sync::{Arc, Mutex};
+use toml::Value;
+use std::fs;
+use std::{thread, time};
+
+const DEFAULT_CONFIG_FILE: &str = "flowgger.toml";
+const DEFAULT_OUTPUT_FILEPATH: &str = "output.log";
+const DEFAULT_QUEUE_SIZE: usize = 10_000_000;
+
+const DEFAULT_OUTPUT_FORMAT: &str = "gelf";
+const DEFAULT_OUTPUT_FRAMING: &str = "noop";
+const DEFAULT_OUTPUT_TYPE: &str = "file";
+
+const DEFAULT_FUZZED_MESSAGE_COUNT: u64 = 500;
+
+fn get_file_output(config: &Config) -> Box<dyn Output> {
```

Comment:
there is not a single comment in the file. please explain what you're doing, why, what are the success criteria, when is this run....